calico.go

```diff
@@ -44,80 +44,91 @@ func init() {
 	// since namespace ops (unshare, setns) are done for a single thread, we
 	// must ensure that the goroutine does not jump from OS thread to thread
 	runtime.LockOSThread()
-
-	nodename, _ = os.Hostname()
-}
-
-func updateNodename(conf NetConf, logger *log.Entry) {
-	if conf.Hostname != "" {
-		nodename = conf.Hostname
-		logger.Warn("Configuration option 'hostname' is deprecated, use 'nodename' instead.")
-	}
-	if conf.Nodename != "" {
-		nodename = conf.Nodename
-	}
 }
 
 func cmdAdd(args *skel.CmdArgs) error {
 	// Unmarshal the network config, and perform validation
-	conf := NetConf{}
+	conf := types.NetConf{}
 	if err := json.Unmarshal(args.StdinData, &conf); err != nil {
 		return fmt.Errorf("failed to load netconf: %v", err)
 	}
 
-	cniVersion := conf.CNIVersion
+	utils.ConfigureLogging(conf.LogLevel)
 
-	ConfigureLogging(conf.LogLevel)
+	// Allow the nodename to be overridden by the network config
+	nodename = utils.DetermineNodename(conf)
 
-	workload, orchestrator, err := GetIdentifiers(args)
+	// Extract WEP identifiers such as pod name, pod namespace (for k8s), containerID, workload name.
+	epIDs, err := utils.GetIdentifiers(args, nodename)
 	if err != nil {
 		return err
 	}
 
-	logger := CreateContextLogger(workload)
-
-	// Allow the nodename to be overridden by the network config
-	updateNodename(conf, logger)
-
-	logger.WithFields(log.Fields{
-		"Orchestrator": orchestrator,
-		"Node":         nodename,
-		"Workload":     workload,
-		"ContainerID":  args.ContainerID,
-	}).Info("Extracted identifiers")
+	logger := &log.Entry{}
+	if epIDs.Orchestrator == "k8s" {
+		logger = log.WithFields(log.Fields{
+			"WorkloadEndpoint": epIDs.Workload,
```

Comment:
What is the Workload for a k8s orchestrator, I think I asked that before but got shouted down.  Shouldn't it be blank?

We'll have a Pod, Namespace, Endpoint, Node that uniquely identifier the WEP in k8s no?  The Workload is not required ... unless I'm misunderstanding.  Let's chat.

Reply:
and ContainerID as an additional identifier too of course.

Reply:
@robbrockbank we might be talking in circles about the same thing here.

Basically, we just need logs to be emitted which include the following information:

- WorkloadEndpoint Name
- The ContainerID of the container being networked.
- The Kubernetes pod Name and Namespace pair.